tests/shardddl/run.sh

```diff
@@ -0,0 +1,1586 @@
+    run_sql_source2 "insert into ${shardddl1}.${tb2} (id) values(6);"
+    sleep 1
+    run_sql_source2 "alter table ${shardddl1}.${tb1} change a new_col datetime default now();"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} (id) values(7);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} (id) values(8);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} (id) values(9);"
+    sleep 1
+    run_sql_source2 "alter table ${shardddl1}.${tb2} change a new_col datetime default now();"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} (id) values(10);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} (id) values(11);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} (id) values(12);"
+    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+}
+
+function DM_059() {
+    run_case 059 "double-source-pessimistic" \
+    "run_sql_source1 \"create table ${shardddl1}.${tb1} (id int, a datetime);\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb1} (id int, a datetime);\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb2} (id int, a datetime);\"" \
+    "clean_table" "optimistic"
+    run_case 059 "double-source-pessimistic" \
+    "run_sql_source1 \"create table ${shardddl1}.${tb1} (id int, a datetime);\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb1} (id int, a datetime);\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb2} (id int, a datetime);\"" \
+    "clean_table" "optimistic"
+}
+
+function DM_062_CASE() {
+    run_sql_source1 "alter table ${shardddl1}.${tb1} modify id bigint;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(1);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(2);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(3);"
+    run_sql_source2 "alter table ${shardddl1}.${tb1} modify id bigint;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(4);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(5);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(6);"
+    run_sql_source2 "alter table ${shardddl1}.${tb2} modify id bigint;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(7);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(8);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(9);"
+    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+}
+
+function DM_062() {
+    run_case 062 "double-source-pessimistic" "init_table 111 211 212" "clean_table" "pessimistic"
+    run_case 062 "double-source-optimistic" "init_table 111 211 212" "clean_table" "optimistic"
+}
+
+function DM_063_CASE() {
+    run_sql_source1 "alter table ${shardddl1}.${tb1} modify id int(15);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(1);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(2);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(3);"
+    run_sql_source2 "alter table ${shardddl1}.${tb1} modify id int(20);"
```

Comment:
It seems not to match the original designed case.

Reply:
address it in e4fd8d5. TiDB currently not support modify int to float, use small int instead.

Reply:
It seems that we currently can not detect conflict for `modify column mediumint/bigint` in optimistic mode. Have added TODO comment and will open issure to track it.